swap/peer.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package swap
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	contract "github.com/ethersphere/swarm/contracts/swap"
+	"github.com/ethersphere/swarm/p2p/protocols"
+)
+
+// ErrDontOwe indictates that no balance is actially owned
+var ErrDontOwe = errors.New("no negative balance")
+
+// Peer is a devp2p peer for the Swap protocol
+type Peer struct {
+	*protocols.Peer
+	swap               *Swap
```

Comment:
we dont need swap as a field in peer. 
we only need statestore if balance and cheque accessors are defined on Peer

Reply:
can we postpone this post merge?

Reply:
i don't see any methods on peer that require peer to have `swap`. are you defining methods on the type outside of this file that require this? message handlers were already refactored to be on `Swap` and so this can probably be safely removed